channeldb/graph.go

```diff
@@ -1304,6 +1312,46 @@ func putLightningNode(nodeBucket *bolt.Bucket, aliasBucket *bolt.Bucket, node *L
 		return err
 	}
 
+	if err := node.Features.Encode(&b); err != nil {
+		return err
+	}
+
+	numAddresses := uint16(len(node.Addresses))
+	byteOrder.PutUint16(scratch[:2], numAddresses)
+	if _, err := b.Write(scratch[:2]); err != nil {
+		return err
+	}
+
+	for _, address := range node.Addresses {
+		if address.Network() == "tcp" {
+			if address.(*net.TCPAddr).IP.To4() != nil {
+				scratch[0] = 0
+				if _, err := b.Write(scratch[:1]); err != nil {
+					return err
+				}
+
+				copy(scratch[:4], address.(*net.TCPAddr).IP.To4())
```

Comment:
From my reading, this looks like it'll override the address type which is the first element in the slice. To fix this, the bounds on this write should instead be:
```go
scratch[1:5]
```